sw_i2c/sensirion_sw_i2c.c

```diff
@@ -125,17 +147,21 @@ s8 sensirion_i2c_read(u8 address, u8* data, u16 count)
 {
     s8 ret;
     u16 i;
-    sensirion_i2c_start();
+
+    ret = sensirion_i2c_start();
+    if (ret != STATUS_OK)
+        return ret;
+
     ret = sensirion_i2c_write_byte((address << 1) | 1);
     if (ret != STATUS_OK) {
         sensirion_i2c_stop();
         return ret;
     }
     for (i = 0; i < count; i++)
-        data[i] = sensirion_i2c_read_byte(1);
+        data[i] = sensirion_i2c_read_byte(i < count - 1);
```

Comment:
This is a bit too magic for my taste. Maybe introduce a temporary variable `u8 send_ack = i < (count - 1);`?